Content.Server/Gatherable/GatherableSystem.cs

```diff
@@ -42,16 +45,22 @@ private void OnInteractUsing(EntityUid uid, GatherableComponent component, Inter
 
         cancelToken = new CancellationTokenSource();
         tool.GatheringEntities[uid] = cancelToken;
+        var multiplier = FixedPoint2.New(1f);
 
-        var doAfter = new DoAfterEventArgs(args.User, tool.GatheringTime, cancelToken.Token, uid)
+        if (TryComp<DestructibleComponent>(uid, out var destructible))
+        {
+            multiplier = _destructible.DestroyedAt(uid, destructible) / tool.Damage.Total;
```

Comment:
- Maybe add a check in case tool.Damage.Total == 0
- `DestroyedAt` can return `FixedPoint2.MaxValue` if the entity isn't destructible. Maybe add a check for that with some error.
- If the object is not susceptible to all of the damage types, this won't actually be enough to kill it.
  - If you want it to be accurate, you have to iterate over damage types in the tool and check if they apply to the target. If it does, add it to a separately tracked total tool damage